auctions/english-auction/project/contracts/auction-contract/tests/functions/core/bid.rs

```diff
@@ -1,493 +1,234 @@
 use crate::utils::{
-    interface::core::{
-        auction::{bid, create},
-        nft::mint,
-    },
-    setup::{defaults_nft, defaults_token, nft_asset, setup, token_asset},
+    interface::core::auction::{bid, create},
+    setup::{defaults, setup},
 };
 use fuels::types::Identity;
 
 mod success {
 
     use super::*;
     use crate::utils::{
-        interface::{
-            core::nft::{approve, mint, set_approval_for_all},
-            info::{auction_info, deposit_balance},
-        },
-        setup::{Auction, AuctionAsset, State},
+        interface::info::{auction_info, deposit_balance},
+        setup::{Auction, State},
     };
 
     #[tokio::test]
     async fn overrides_bid() {
-        let (_, seller, buyer1, buyer2, _, sell_token_contract_id, _, buy_token_contract_id, _) =
-            setup().await;
+        let (_, seller, buyer1, buyer2, _, sell_asset, buy_asset) = setup().await;
         let (sell_amount, initial_price, reserve_price, duration, _initial_wallet_amount) =
-            defaults_token().await;
+            defaults().await;
 
         let seller_identity = Identity::Address(seller.wallet.address().into());
         let buyer1_identity = Identity::Address(buyer1.wallet.address().into());
         let buyer2_identity = Identity::Address(buyer2.wallet.address().into());
-        let sell_asset = token_asset(sell_token_contract_id, sell_amount).await;
-        let buy_asset = token_asset(buy_token_contract_id, 0).await;
-        let bid_asset = token_asset(buy_token_contract_id, initial_price).await;
-        let bid2_asset = token_asset(buy_token_contract_id, initial_price + 1).await;
 
         let auction_id = create(
-            buy_asset.clone(),
+            buy_asset,
             &seller.auction,
             duration,
             initial_price,
             Some(reserve_price),
             seller_identity.clone(),
-            sell_asset.clone(),
+            sell_asset,
+            sell_amount,
         )
         .await;
 
         let buyer1_deposit =
             deposit_balance(auction_id, &buyer1.auction, buyer1_identity.clone()).await;
         assert!(buyer1_deposit.is_none());
 
-        bid(auction_id, bid_asset.clone(), &buyer1.auction).await;
+        bid(auction_id, buy_asset, initial_price, &buyer1.auction).await;
 
-        let buyer1_deposit: AuctionAsset =
-            deposit_balance(auction_id, &buyer1.auction, buyer1_identity.clone())
-                .await
-                .unwrap();
+        let buyer1_deposit = deposit_balance(auction_id, &buyer1.auction, buyer1_identity.clone())
+            .await
+            .unwrap();
         let auction: Auction = auction_info(auction_id, &seller.auction).await.unwrap();
-        assert_eq!(buyer1_deposit, bid_asset);
-        assert_eq!(auction.bid_asset, bid_asset);
+        assert_eq!(buyer1_deposit, initial_price);
```

Comment:
We should continue to enforce the asset in these tests

Reply:
Okay but we do need to enforce that the contract didn't change the asset and thus will send the correct reward. That's the point of testing. To internally lock down the components to expected behaviour instead of trusting the contract